### ISSUE TITLE
Train on apache hadoop yarn takes more time as the worker >=2 in configuration

### ISSUE THREAD

**Issue by wallyell** [dmlc/wormhole#71]
It seems that the train speed is slower as the worker(node) in configuration changed from 1 to 2 or more.Can anyone tell me why this happens, is there anything wrong with my configuration?

**The configuration：**

booster = gbtree
objective = multi:softmax
eta = 0.5
max_depth = 5
num_class = 10
num_round = 50
save_period = 0
eval_train = 1

**The Shell Script:**

../../dmlc-core/tracker/dmlc-submit --cluster=yarn --num-workers=4 --worker-cores=2
    ../../xgboost parameter.conf nthread=16\
    data=hdfs://hadoop01:8020/xgb-demo/train\
    eval[test]=hdfs://hadoop01:8020/xgb-demo/test\
    model_dir=hdfs://hadoop01:8020/xgb-demo/model
